src/amo/components/Addon/index.js

```diff
@@ -173,114 +185,174 @@ export class AddonBase extends React.Component {
-    // TODO: We should file an API bug about this...
-    const summary = addon.summary ? addon.summary : addon.description;
-
-    const title = i18n.sprintf(
-      // L10n: Example: The Add-On <span>by The Author</span>
-      i18n.gettext('%(addonName)s %(startSpan)sby %(authorList)s%(endSpan)s'), {
-        addonName: addon.name,
-        authorList: authorList.join(', '),
-        startSpan: '<span class="Addon-author">',
-        endSpan: '</span>',
-      });
+    let errorBanner = null;
+    if (errorHandler.hasError()) {
+      log.error('Captured API Error:', errorHandler.capturedError);
+      if (errorHandler.capturedError.responseStatusCode === 404) {
+        return <NotFound />;
+      }
+      // Show a list of errors at the top of the add-on section.
+      errorBanner = errorHandler.renderError();
+    }
 
-    const {
-      compatible, maxVersion, minVersion, reason,
-    } = getClientCompatibility({ addon, clientApp, userAgentInfo });
+    const addonType = addon ? addon.type : ADDON_TYPE_EXTENSION;
 
-    // eslint-disable react/no-danger
-    return (
-      <div className={classNames('Addon', `Addon-${addon.type}`)}>
-        <Card className="" photonStyle>
-          <header className="Addon-header">
-            <h1
-              className="Addon-title"
-              dangerouslySetInnerHTML={sanitizeHTML(title, ['a', 'span'])}
-            />
+    let authorList = [<LoadingText key="LoadingText" />];
+    if (addon) {
+      authorList = addon.authors.map(
+        (author) => `<a href="${author.url}">${author.name}</a>`);
+    }
 
-            <p
-              className="Addon-summary"
-              dangerouslySetInnerHTML={sanitizeHTML(summary, ['a'])}
-            />
+    const summaryProps = {};
+    let summaryPlaceholder = null;
+    if (addon) {
+      // Themes lack a summary so we do the inverse :-/
+      // TODO: We should file an API bug about this...
+      const summary = addon.summary ? addon.summary : addon.description;
+      summaryProps.dangerouslySetInnerHTML = sanitizeHTML(summary, ['a']);
+    } else {
+      summaryPlaceholder = <LoadingText />;
+    }
 
-            <InstallButton
-              {...this.props}
-              className="Button--action Button--small"
-              disabled={!compatible}
-              ref={(ref) => { this.installButton = ref; }}
-              status={installStatus}
-            />
+    const titleProps = {};
+    let titlePlaceholder = null;
+    if (addon) {
+      const title = i18n.sprintf(
+        // L10n: Example: The Add-On <span>by The Author</span>
+        i18n.gettext('%(addonName)s %(startSpan)sby %(authorList)s%(endSpan)s'), {
+          addonName: addon.name,
+          authorList: authorList.join(', '),
+          startSpan: '<span class="Addon-author">',
+          endSpan: '</span>',
+        }
+      );
+      titleProps.dangerouslySetInnerHTML = sanitizeHTML(title, ['a', 'span']);
+    } else {
+      titlePlaceholder = <LoadingText />;
```

Comment:
I tried locally and I think this would benefit from a higher `minWidth` which is a property on `<LoadingText />` – it sets the minimum % the text block will fill. As titles aren't usually a few letters something like 70% or even 100% might work better for a lot of these short fields.